### PR TITLE
Adding `is:modded` filter

### LINKED ISSUE #3117: Mods 2.0 - Search for gear that has Weapon Mods or Armor Mods inserted
With the introduction of Mods 2.0 the economy of mods and scarcity of what was once a commodity is now a sought after resource that you need to carefully decide which piece of gear should have the honor of consuming your mod resource.

If you're someone who's been hoarding every drop though, to eventually come back later and look through the perk selection with a fine-toothed comb, you may also want a quick way to parse your inventory to see gear that has mods inserted into them.

Currently, I don't think there's a way to search for weapons, armor, or both for a way to see if they have a mod inserted into them. Of course new weapons, and all Y1/Y2 armor is moddable, but what if you just want to search to see what weapons or armor already have a mod installed?

Search variable ideas include:

`is:modded` (all gear that has a mod installed, maybe unnecessary)
`is:weaponmod` (Highlight weapon mods in inventory, and weapons with a mod installed)
`is:armormod` (Highlight armor mods in inventory, and armor with a mod installed)

This could allow for a quick way for you to find gear you've relegated to your vault, but might not be aware that it has a mod installed in it. Since the mods return to inventory every time you dismantle a piece of gear with them, this could expedite your search to reclaim lost mods.

If there's a way to search or parse this with DIM I may have glossed over it and couldn't find it when looking through search options. I mainly just wanted to present the idea as it doesn't appear to exist in this implementation at least- of course however it may make the most amount of sense to implement such as variable names, etc I'm not married to any ideas and just wanted to get this out there.

Thanks to everyone on the DIM team for a great and helpful management site

### PR DESCRIPTION
**Pull request by kyleshay**
Looks for the mod slot and filters out ones that are empty

This resolves #3117

### CODE REVIEW

File: src/app/search/search-filters.ts [hunk 1]
```diff
@@ -1100,6 +1101,18 @@ export function searchFilters(
...
         );
       },
+      hasMod(item: D2Item) {
+        return (
+          item.sockets &&
+          item.sockets.sockets.some((socket) => {
+            return !!(
+              (socket.plug || false) &&
+              ![2323986101, 2600899007].includes(socket.plug.plugItem.hash) &&
+              socket.plug.plugItem.itemTypeDisplayName.includes(' Mod')
```
Comment (kyleshay) [inline src/app/search/search-filters.ts]: Is there a better thing to look for? :s

Comment (delphiactual) [reply src/app/search/search-filters.ts]: will this work with i18n?

Comment (delphiactual) [reply src/app/search/search-filters.ts]: itemType=19?
https://data.destinysets.com/i/InventoryItem:444600262

Comment (delphiactual) [reply src/app/search/search-filters.ts]: ItemCategory:59

Comment (kyleshay) [reply src/app/search/search-filters.ts]: Hmm, itemCategory part of `socket.plug.plugItem.itemCategoryHashes.includes(59)` seems to match all equippable items. The rest of the itemCategoryHashes don't seem to be any more useful. One is for the specific bucket, the other for weapon/armor sections. 

I've found that similar to masterwork, the `plugCategoryIdentifier` has a somewhat unique string id that I can use.

**Pull request merged**